src/setup/test/serverless-config_test.go

```diff
@@ -1,23 +1,110 @@
 package setup
 
 import (
+	"bytes"
 	"github.com/stretchr/testify/require"
+	"os"
 	"stellar/setup"
 	"testing"
 )
 
+func TestCreateHeaderConfig(t *testing.T) {
+
+	// Define the expected Serverless struct and its associated values
+	expected := &setup.Serverless{
+		Service:          "STeLLAR",
+		FrameworkVersion: "3",
+		Provider: setup.Provider{
+			Name:    "aws",
+			Runtime: "go1.x",
+			Region:  "us-west-1",
+		},
+	}
+
+	// Define the Configuration struct for testing
+	config := &setup.Configuration{
+		Provider: "aws",
+		Runtime:  "go1.x",
+	}
+
+	actual := &setup.Serverless{}
+	actual.CreateHeaderConfig(config)
+
+	require.Equal(t, expected, actual)
+}
+
 func TestAddFunctionConfig(t *testing.T) {
-	resultServerless := &setup.Serverless{}
-	requireServerless := &setup.Serverless{}
-	subEx := &setup.SubExperiment{Title: "test1"}
-	resultServerless.AddFunctionConfig(subEx, 0)
+	expected := &setup.Serverless{
+		Package: setup.Package{Patterns: []string{"pattern1"}},
+		Functions: map[string]*setup.Function{
+			"test1_2_0": {
+				Name:    "test1_2_0",
+				Handler: "hellopy/lambda_function.lambda_handler",
+				Runtime: "Python3.8",
+				Events: []setup.Event{
+					{setup.HttpApi{Path: "/test1_2_0", Method: "GET"}}}},
```

Comment:
The linter might be failing because the key for HttpApi was not specified.
```diff
- {setup.HttpApi{Path: "/test1_2_0", Method: "GET"}}}}
+ {HttpApi: setup.HttpApi{Path: "/test1_2_0", Method: "GET"}}}}
```

Reply:
thanks for the fix!